enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,574 @@
+$ cd ..
+$ opm index update community-operators --tag=docker.io/my-namespace/community-operators:latest
+$ rm -rf community-operators
+$ opm index inspect packages --index=docker.io/my-namespace/community-operators --output=json
+$ cat community-operators/etcd.json
+"etcd": {
+        "name": "etcd",
+        "bundles": [
+            {
+                "path": "quay.io/random/etcdv0.0.1",
+                "version": "v0.0.1",
+                "channels": ["alpha"],
+            },
+            {
+                "path": "quay.io/somethingelse/etcdv0.0.2",
+                "version": "v0.0.2",
+                "channels": ["beta"],
+            }
+        ],
+        "defaultChannel": "alpha",
+        "icon": "asdfasdfoipuasdfvasdfqer",
+        "channelUpgradeGraphs": {
+            "channels": [
+                {
+                    "name": "alpha",
+                    "upgrades": [
+                        {
+                            "to": {
+                                "bundle": "quay.io/random/etcdv0.0.1"
+                            }
+                        },
+                    ]
+                },
+                {
+                    "name": "alpha",
+                    "upgrades": [
+                        {
+                            "to": {
+                                "bundle": "quay.io/random/etcdv0.0.2"
+                            }
+                        },
+                    ]
+                }
+            ]
+        }
+    }
+
+```
+
+Similarly, removing a `.json` from `community-operators` and pushing that update up to the container image removes the bundle. 
+
+#### Story 3
+
+As a user with pull permission from the namespace an index is hosted in/as a component of operator-lifecycle-manager, I can validate the structure of the content of an index using the json representation.
+ 
+```bash
+$ cd community-operators 
+$ //TODO come up with sed command to remove "defaultchannel":"alpha" from etcd.json
+$ git status 
+	modified:    etcd.json
+
+$ cd ..
+$ opm index validate community-operators
+marshal error: etcd.defaultchannel: cannot convert incomplete value "string" to JSON
+$ opm index update community-operator --tag=docker.io/mynamespace/community-operators:latest
+$ opm index validate docker.io/mynamespace/community-operators:latest 
+marshal error: etcd.defaultchannel: cannot convert incomplete value "string" to JSON
+$ cd community-operators 
+$ git checkout etcd.json
+$ cd ..
+$ opm index validate community-operators 
+No errors found!
+```
+
+#### Story 4 
+
+Given a set of package representations, or an index representations, I can author a new index using just those representations.
+
+```bash
+$ tree community-operators 
+community-operators
+├── amqstreams.json
+└── etcd.json
+└── servicemesh.json
+$ rm community-operators/servicemesh.json
+$ opm index create --from=community-operators --tag=docker.io/someothernamespace/new-community-operators
+```
+### Implementation Details
```

Comment:
So, would not easier we only update the bundle manifests (csv and/or crds) and run a command to overwrite the bundle with new name and/or version and/or channnels? E.g `operator-sdk generate bundle --input-dir=<Directory to read an existing bundle from> --overwrite=true --version=<new version>` and/or `operator-sdk generate bundle --input-dir=<Directory to read an existing bundle from> --overwrite=true --version=<new version> --package=<optional new pkg name> --channels=<optional list of channels>` ?

I am not sure if I got what are the advantages in the approach proposed and what are the use cases that would be solved with this EP.

Reply:
@camilamacedo86 with the background context provided in [this comment](https://github.com/operator-framework/enhancements/pull/37/files/9948060747ac4cd336add31044b75ed55376e1f4#r552058237), does the [motivation of the EP](https://github.com/operator-framework/enhancements/pull/37/files/9948060747ac4cd336add31044b75ed55376e1f4#diff-1edbfba9fdddd2b74b5cd32c6de146a4f62e1400a93bee78aed2cbf6ebaaa1d7R33) make more sense? 
Essentially, we're discussing all these at the index(collection of operators) levels and not at the bundle (single operator) level.